static/app/views/dashboardsV2/gridLayout/dashboard.tsx

```diff
@@ -169,6 +170,18 @@ class Dashboard extends Component<Props> {
     this.props.onUpdate(nextList);
   };
 
+  handleDuplicateWidget = (widget: Widget, index: number) => () => {
+    const {dashboard, handleAddLibraryWidgets} = this.props;
+
+    const widgetCopy = cloneDeep(widget);
+    widgetCopy.id = undefined;
```

Comment:
Can we also add `widgetCopy.tempId = undefined;`? If you try to duplicate a duplicated widget (or new widget) then it can overlap with this view

Reply:
d95a19b527598d6a47194368e73b97b5e99f2011